crates/typst-cli/src/compile.rs

```diff
@@ -199,10 +204,16 @@ fn export_image(
 ) -> StrResult<()> {
     // Determine whether we have a `{n}` numbering.
     let output = command.output();
-    let string = output.to_str().unwrap_or_default();
-    let numbered = string.contains("{n}");
-    if !numbered && document.pages.len() > 1 {
-        bail!("cannot export multiple images without `{{n}}` in output path");
+    let can_handle_multiple = match output {
+        Output::Stdout => false,
+        Output::Path(ref output) => output.to_str().unwrap_or_default().contains("{n}"),
+    };
+    if !can_handle_multiple && document.pages.len() > 1 {
+        let s = match output {
+            Output::Stdout => "to stdout",
```

Comment:
Probably it's possible to write multiple pages to stdout, but I don't know

1. how to do it
2. how user will read multiple images from stdin

Reply:
On 2.: Not sure if this will help, but I once did this with the series of JPEG thumbnails that exiftool can output, using JPEG's end of image (EOI) marker (0xFFD9). PNG seems to have something similar with an "IEND" chunk…

Reply:
I tried searching a bit (unsuccessfully). From your description, I don't think this is a good approach

Reply:
I'd suggest not supporting writing multiple files at first (just error in that case). We can still output multiple pages through PDF.

The alternative would be to use a format such as `tar` to compact multiple files into one, but I don't see that being very convenient at first. However, one possible motivation for a `tar` approach could be to support regular (non-stdout) export to `tar` as well, as that in itself could be convenient.

But I wouldn't go that far in this PR alone.